misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/expression/QualifiedNameReferenceEvaluator.java

```diff
@@ -86,4 +79,44 @@ public BExpressionValue evaluate() throws EvaluationException {
             throw createEvaluationException(INTERNAL_ERROR, syntaxNode.toSourceCode().trim());
         }
     }
+
+    private Optional<BExpressionValue> searchForModuleConstant(String modulePrefix) throws EvaluationException {
+        ModuleSymbol moduleSymbol = resolvedImports.get(modulePrefix).getResolvedSymbol();
+        Optional<ConstantSymbol> constSymbol = moduleSymbol.constants().stream()
+                .filter(constantSymbol -> constantSymbol.getName().isPresent()
+                        && constantSymbol.getName().get().equals(nameReference))
+                .findAny();
+
+        if (constSymbol.isPresent()) {
+            if (!isPublicSymbol(constSymbol.get())) {
+                throw createEvaluationException(NON_PUBLIC_OR_UNDEFINED_ACCESS, nameReference);
+            }
+            Optional<BExpressionValue> moduleVariable = VariableUtils.getModuleVariable(context, moduleSymbol,
+                    nameReference);
+            if (moduleVariable.isEmpty()) {
+                throw createEvaluationException(QUALIFIED_VARIABLE_RESOLVING_FAILED, modulePrefix, nameReference);
+            }
+            return moduleVariable;
+        }
+
+        return Optional.empty();
+    }
+
+    private Optional<BExpressionValue> searchForModuleVariable(ModuleSymbol moduleSymbol, String modulePrefix)
+            throws EvaluationException {
+        Optional<VariableSymbol> variableSymbol = moduleSymbol.allSymbols().stream()
+                .filter(symbol -> symbol.kind() == SymbolKind.VARIABLE
+                        && symbol.getName().isPresent()
+                        && symbol.getName().get().equals(nameReference))
+                .map(symbol -> (VariableSymbol) symbol)
+                .findAny();
+
+        if (variableSymbol.isEmpty()) {
+            throw createEvaluationException(NON_PUBLIC_OR_UNDEFINED_ACCESS, modulePrefix, nameReference);
+        } else if (!isPublicSymbol(variableSymbol.get())) {
+            throw createEvaluationException(NON_PUBLIC_OR_UNDEFINED_ACCESS, modulePrefix, nameReference);
+        }
+
+        return VariableUtils.getModuleVariable(context, variableSymbol.get().getModule().orElseThrow(), nameReference);
```

Comment:
Will `.getModule()` be null or empty value?

Reply:
I was under the impression that module cannot be empty since the symbol is a variable symbol in here. Anyway modified the logic to have a check with 7dc503d